patches/util-linux/0001-Stub-out-several-unsupported-things-on-managarm.patch

```diff
@@ -0,0 +1,175 @@
+@@ -132,4 +133,5 @@ int ul_path_readf_cpuset(struct path_cxt *pc, cpu_set_t **set, int maxcpus, cons
+ int ul_path_readf_cpulist(struct path_cxt *pc, cpu_set_t **set, int maxcpus, const char *path, ...)
+ 				__attribute__ ((__format__ (__printf__, 4, 5)));
+ #endif /* HAVE_CPU_SET_T */
++#endif /* !__managarm__ */
+ #endif /* UTIL_LINUX_PATH_H */
+diff --git a/lib/cpuset.c b/lib/cpuset.c
+index 2847db853..78ffd5f1b 100644
+--- a/lib/cpuset.c
++++ b/lib/cpuset.c
+@@ -1,3 +1,4 @@
++#ifndef __managarm__
+ /*
+  * Terminology:
+  *
+@@ -411,3 +412,4 @@ usage_err:
+ 	exit(EXIT_FAILURE);
+ }
+ #endif
++#endif
+diff --git a/lib/path.c b/lib/path.c
+index 75fa85305..fe3702970 100644
+--- a/lib/path.c
++++ b/lib/path.c
+@@ -977,6 +977,7 @@ FILE *ul_prefix_fopen(const char *prefix, const char *path, const char *mode)
+ 	return fopen(buf, mode);
+ }
+ 
++#ifndef __managarm__
+ #ifdef HAVE_CPU_SET_T
+ static int ul_path_cpuparse(struct path_cxt *pc, cpu_set_t **set, int maxcpus, int islist, const char *path, va_list ap)
+ {
+@@ -1044,6 +1045,7 @@ int ul_path_readf_cpulist(struct path_cxt *pc, cpu_set_t **set, int maxcpus, con
+ }
+ 
+ #endif /* HAVE_CPU_SET_T */
++#endif /* !__managarm__ */
+ 
+ 
+ #ifdef TEST_PROGRAM_PATH
+diff --git a/libblkid/src/topology/ioctl.c b/libblkid/src/topology/ioctl.c
+index 3aba09e4f..76b750e20 100644
+--- a/libblkid/src/topology/ioctl.c
++++ b/libblkid/src/topology/ioctl.c
+@@ -16,6 +16,8 @@
+ #include <unistd.h>
+ #include <errno.h>
+ 
++#include <linux/fs.h>
++
+ #include "topology.h"
+ 
+ /*
+diff --git a/login-utils/logindefs.c b/login-utils/logindefs.c
+index 97150dc28..35371ec3f 100644
+--- a/login-utils/logindefs.c
++++ b/login-utils/logindefs.c
+@@ -26,7 +26,7 @@
+ #include <stdio.h>
+ #include <stdlib.h>
+ #include <string.h>
+-#include <sys/syslog.h>
++#include <syslog.h>
+ #include <sys/stat.h>
+ #include <sys/types.h>
+ #include <pwd.h>
+diff --git a/login-utils/lslogins.c b/login-utils/lslogins.c
+index 8a6ef713f..2b56879c3 100644
+--- a/login-utils/lslogins.c
++++ b/login-utils/lslogins.c
+@@ -25,7 +25,7 @@
+ #include <getopt.h>
+ #include <sys/types.h>
+ #include <sys/stat.h>
+-#include <sys/syslog.h>
++#include <syslog.h>
+ #include <pwd.h>
+ #include <grp.h>
+ #include <shadow.h>
+diff --git a/misc-utils/lsblk.h b/misc-utils/lsblk.h
+index 87221271e..6ae1ebecc 100644
+--- a/misc-utils/lsblk.h
++++ b/misc-utils/lsblk.h
+@@ -13,6 +13,8 @@
+ 
+ #include <libsmartcols.h>
+ 
++#include <linux/fs.h>
++
+ #include "c.h"
+ #include "list.h"
+ #include "debug.h"
+diff --git a/term-utils/agetty.c b/term-utils/agetty.c
+index 191fa29c0..bf985a2f9 100644
+--- a/term-utils/agetty.c
++++ b/term-utils/agetty.c
+@@ -88,6 +88,14 @@
+ #  ifndef DEFAULT_STERM
+ #    define DEFAULT_STERM  "vt102"
+ #  endif
++#elif defined __managarm__
++#  define USE_SYSLOG
++#  ifndef DEFAULT_VCTERM
++#    define DEFAULT_VCTERM "managarm"
++#  endif
```

Comment:
Is this a terminal name? If yes, I don't think we should define a "managarm" terminal type.

Reply:
I think it indeed it. I can change it to match Linux there, that may be more appropriate.

Reply:
Agreed.